addons/mass_mailing/wizard/mailing_contact_to_list.py

```diff
@@ -30,21 +30,23 @@ def action_add_contacts_and_send_mailing(self):
     def _add_contacts_to_mailing_list(self, action):
         self.ensure_one()
 
-        previous_count = len(self.mailing_list_id.contact_ids)
+        contacts_to_add = self.contact_ids.filtered(lambda c: c.id not in self.mailing_list_id.contact_ids.ids)
         self.mailing_list_id.write({
-            'contact_ids': [
-                (4, contact.id)
-                for contact in self.contact_ids
-                if contact not in self.mailing_list_id.contact_ids]
-            })
+            'subscription_ids': [
+                Command.create({
```

Comment:
Command is outdated, please use
```suggestion
                (0, 0, {
```